lib/gmd-api-caller.ts

```diff
@@ -0,0 +1,51 @@
+import axios, { AxiosRequestConfig } from './get-axios.js';
+export type Logger = (msg: string) => void;
+
+export class RemoteAPICaller {
+    baseURL: URL;
+    log: Logger | null;
+    constructor(baseURL: URL) {
+        this.baseURL = baseURL;
+        this.log = null;
+    }
+
+    setLogger(logger: Logger) {
+        this.log = logger;
+    }
+
+    /**
+    *  API call to a GMD node. This call is done over network.
+    *
+    *
+    * @param {String} method HTTP method (only 'get' or 'post' are used)
+    * @param {JSON} params *  Full set of API methods and parameters can be seen here: https://node.thecoopnetwork.io/test.
+    *  All parameters will be passed as a single json via 'params' parameter. Exception to this is 'secretPhrase' parameter which
+    * should not be included in params, and even if you include it, the SDK will delete it before making the API call to the node.
+    *  In addition to all parameters described above there are the following parameters:
+    *  'requestType': [mandatory] wich is the name of the GMD API method (e.g. 'sendMoney', 'sendMessage', 'getPolls' etc..)
+    *  'baseURL': [optional] URL of the GMD node where this request is performed. By default https://node.thecoopnetwork.io main net is used.
+    *  'httpTimeout' [optional] parameter for HTTP request to specify a timeout when GMD node not reachable. Axios default is used if this param is not specified.
+    *  Example:
+    *  params = {
+            requestType: 'getAccountsBulk',
+            pageSize: 3,
+            page: 0,
+            baseURL: 'https://node.thecoopnetwork.io:6877'
+        }
+    * @returns {Promise} that will resolve to the body of the server response (usually a JSON).
+    */
+    async apiCall(method: string, params: Record<string, string>) {

```

Comment:
You could use the async annotation here for consistency (and a return type). Do we know ahead of type what data can be? It's super nice you made it unknown, but we could make this function generic

```
async apiCall<T = unknown>(): Promise<T>
```

if the called knows what to expect.

LE: Yes, make it generic, the method can also be `method: "get" | "post"` to be more typesafe

Reply:
1. I could not use await on the axios call as I need to cover the case when user wants to make multiple call to the API at the same time. Putting await on axios call effectively will be serializing the call to the API. By returning a Promise I leave it to the use to make as many calls as he wants, collecting all the promises and that awaiting with `await Promise.all()`.

2. The return type of apiCall() more specific than Record<String, unknown> cannot be  known here as this is a call to a remote GMD node and the response is specific to that endpoint.

3. Added method: "get" | "post". Thank you for suggestion.